src/script/descriptor.cpp

```diff
@@ -297,6 +299,69 @@ class ConstPubkeyProvider final : public PubkeyProvider
     }
 };
 
+/** An object representing a parsed list of pubkeys in a descriptor. */
+class ListPubkeyProvider final : public PubkeyProvider
+{
+    std::vector<std::unique_ptr<PubkeyProvider>> m_pubkeys;
+
+public:
+    ListPubkeyProvider(uint32_t exp_index, std::vector<std::unique_ptr<PubkeyProvider>> pubkeys) : PubkeyProvider(exp_index), m_pubkeys(std::move(pubkeys)) {}
+    bool GetPubKey(int pos, const SigningProvider& arg, CPubKey& key, KeyOriginInfo& info, const DescriptorCache* read_cache = nullptr, DescriptorCache* write_cache = nullptr) const override
+    {
+        if ((size_t)pos >= m_pubkeys.size()) {
+            return false;
+        }
+        return m_pubkeys.at(pos)->GetPubKey(0, arg, key, info, read_cache, write_cache);
+    }
+    bool IsRange() const override { return true; }
+    size_t GetSize() const override
+    {
+        return std::accumulate(
+            m_pubkeys.begin(), m_pubkeys.end(),
+            0,
+            [](const size_t acc, const std::unique_ptr<PubkeyProvider>& pubkey) {
+                return std::max(acc, pubkey->GetSize());
```

Comment:
If there are mixed compressed and uncompressed keys in the list, then this will always use the uncompressed size, which may result in overestimating the size. I think this will only be used by miniscript though.